TAScheduler/UnitTests/tests.py

```diff
@@ -11,38 +13,165 @@
 # Feel free to make suggestions on discord (add/remove/edit methods)!.
 ### Rememeber: These methods were made before any coding (I was guessing) so it's likely they should be changed.
 class TestUserLogin(TestCase):  # Alec
-    pass
-
-
-class TestUserGetID(TestCase):  # Alec
-    pass
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address="test@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
 
+    def test_login_valid_credentials(self):
+        result = self.user.login("test@example.com", "password123")
+        self.assertTrue(result)
 
-class TestUserGetPassword(TestCase):  # Alec
-    pass
-
-
-class TestUserGetName(TestCase):  # Alec
-    pass
-
+    def test_login_invalid_credentials(self):
+        result = self.user.login("test@example.com", "wrongpassword")
+        self.assertFalse(result)
 
-class TestUserGetRole(TestCase):  # Alec
-    pass
 
+class TestUserGetID(TestCase):  # Alec
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address="test@example.com",
+            password="password123",  # Hashing the password
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+    def test_get_id(self):
+        user_id = self.user.getID()
+        self.assertEqual(user_id, "test@example.com")
+
+
+class TestUserGetPassword(TestCase):
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address="testuser@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+    def test_get_password(self):
+        self.assertEqual(self.user.get_password(), "password123")
+
+class TestUserGetName(TestCase):
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address="testuser@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+
+    def test_get_name(self):
+        self.assertEqual(self.user.get_name(), "Test User")
+
+
+class TestUserGetRole(TestCase):
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address="testuser@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+
+    def test_get_role(self):
```

Comment:
Because you need to use TA/admin/instructor to run this method the role should be ta/admin/instructor

Reply:
good call, forgot about that